src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -1020,6 +1023,129 @@ void MeshForwarder::UpdateRoutes(uint8_t *           aFrame,
     return;
 }
 
+void MeshForwarder::UpdateFragmentLifetime(void)
+{
+    bool shouldRun = false;
+
+    for (size_t i = 0; i < OT_ARRAY_LENGTH(mFragmentEntries); i++)
+    {
+        if (mFragmentEntries[i].GetLifetime() != 0)
+        {
+            mFragmentEntries[i].SetLifetime(mFragmentEntries[i].GetLifetime() - 1);
```

Comment:
suggestion: how about adding `DecrementLifetime()` method to `FragmentPriorityEntry` to do this.

Reply:
Changed.